### PR TITLE
Increment minor version when release is created

### PR DESCRIPTION
**Pull request by petrzpav**
According to issue #31…
- Add test `merge master hotfix (release branch exists)`
- Update `create release` test
- Update `gf_merge_hotfix` and `gf_release` functions
- Improve `merge_branches` function to proceeding merge only on branches with differences
- Add `merge_branches` 4th param to not checkout after merge
- Improve `delete_gf_branch` to handle deleting current branch

### CODE REVIEW

**Comment by petrzpav** [comment]
I found major bug in this PR.. WIP..

**Comment by petrzpav** [comment]
Fixed :feelsgood: … Please review

File: test/test.butt [hunk 1]
```diff
@@ -647,6 +647,43 @@ start "merge master hotfix"
...
+   && git add . \
+   && git commit -m "add file a" \
+   && git_checkout dev \
+   && echo b > b \
+   && git add . \
+   && git commit -m "add file b" \
+   && omgf --yes release \
+   && git_checkout hotfix-myhotfix
+  } >/dev/null
+  debug "echo '' | omgf -v --yes"
```
Comment (jnv) [inline test/test.butt]: Comment

File: test/test.butt [hunk 2]
```diff
@@ -647,6 +647,43 @@ start "merge master hotfix"
...
+  assert_equal $status 5
+  # resolve release conflicts
+  { git add . \
+    && git commit -m "Merge branch 'hotfix-myhotfix' into release" \
+    && git_checkout hotfix-myhotfix
+  } >/dev/null
+  debug "echo '' | omgf -v --yes"
```
Comment (jnv) [inline test/test.butt]: Assert status == 0

**Pull request merged**